src/detect-asn1.h

```diff
@@ -36,12 +36,11 @@
 typedef struct DetectAsn1Data_ {
     uint8_t flags;     /* flags indicating the checks loaded */
     uint32_t oversize_length;   /* Length argument if needed */
-    int32_t absolute_offset;   /* Length argument if needed */
+    uint16_t absolute_offset;   /* Length argument if needed */
```

Comment:
What the rationale for changing the type?
Why leave the type of `relative_offset` unchanged?

Reply:
absolute_offset: Represents an offset from [0, payload_len], `Packet::payload_len` is represented as a `uint16_t`, so [0, uint16::max]
relative_offset: Represents an offset relative to the last content match which is between [0, uint16::max] so relative to this, we need a type to hold the range [-uint16::max, uint16::max] which is why it's int32, with a bounds check on parsing and also contains overflow checks

That was my train of thought, open to suggestions :)

Reply:
Makes sense to me.